csharp/src/Ice/TcpEndpoint.cs

```diff
@@ -13,12 +13,13 @@ namespace ZeroC.Ice
     /// <summary>The Endpoint class for the TCP transport.</summary>
     internal class TcpEndpoint : IPEndpoint
     {
-        public override bool HasCompressionFlag { get; }
         public override bool IsDatagram => false;
         public override bool IsSecure => Transport == Transport.SSL;
-        public override TimeSpan Timeout { get; } = DefaultTimeout;
         public override Transport Transport { get; }
 
+        protected readonly bool HasCompressionFlag;
```

Comment:
I would keep them as get-only `private protected` properties.

Reply:
I guess it should be private, because the class is sealed